test/regression/cluster.py

```diff
@@ -115,6 +107,34 @@ def generate_solution_file(self):
                 elif image.startswith('cortx'):
                     images[image] = cortx_ver['cortx_all']
 
+        #
+        # Set secrets
+        #
+        secrets = self.cluster_data.get('secrets')
+        if secrets:
+            solution['solution']['secrets'] = secrets
+
+        #
+        # Set nodeports
+        #
```

Comment:
This isn't beautiful, but it allows any or all fields in "nodePorts" to be set.

Reply:
Since all of the nodePorts settings are optional, why do you need to go through with all of this?

Reply:
I am defining a minimal config file so that I can easily generate a solution.yaml file.  I don't want to have to specify the parts that don't change -- only the node ports.  (I admit that I'm not sure that I need this, but I will use it for now.  I don't want to use default node ports.).

Reply:
Ok, I think it threw me off because you're defining some of the ports (as `None`), even when they aren't specified by the minimal config.

If you're not validating the schema, you could just do:

```python
extsvc = solution['solution']['common']['external_services']
extsvc.update(nodeports)
```

I guess I'm not seeing the value of `new_nodeports` either way. Why not just modify the solution config as you go?

```python
extsvc = solution['solution']['common']['external_services']
if nodeports:
    if 'control' in nodeports and 'https' in nodeports['control']:
        extsvc.update({'control': { 'https': nodeports['control']['https']})
    etc
```